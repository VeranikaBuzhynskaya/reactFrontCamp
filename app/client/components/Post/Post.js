import React from 'react';
import styles from './post.css';
import {connect} from 'react-redux';
import { removePost } from "../../actions/actions";
import {withRouter} from 'react-router-dom';
import { Link } from 'react-router-dom'

class Post extends React.Component{
    constructor(props){
        super(props);

    }

    render(){
        const info = this.props.info;
        return (
            <article className="poster">
                <h3 className="title">{info.title}</h3>
                <input type="button" value="Delete" className="delete-button-post" 
                onClick={() => {this.props.fetchRemovePosts(info.id)}}/>
                <p className="description">{info.description}</p>
                <div className="post-inform">
                    <span className="data">{info.releaseDate}</span>
                    <span className="author">{info.author}</span>
                </div>
            </article>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
      fetchRemovePosts: (id) => {
          dispatch(removePost(id))
      }
  };
}

export default withRouter(connect(null,mapDispatchToProps)(Post));;
