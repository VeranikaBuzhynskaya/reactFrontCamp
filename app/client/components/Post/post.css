.delete-button-post{
    color: rgb(223, 128, 128);
    padding: 5px 10px;
    background-color: white;
    font-size: 12px;
    border: none;
    cursor: pointer;
    position: absolute;
    right: 5px;
    top: 5px;
    border-radius: 5px;
}

.post-inform .author{
    color: rgb(140,140,140);
}

.post-inform{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 10px 0 5px 0;
}

.post-inform span:first-child{
    font-size: 12px;
    width: 100px;
    font-weight: bold;
    text-transform: uppercase;
}

.title{
    max-width: 290px;
}