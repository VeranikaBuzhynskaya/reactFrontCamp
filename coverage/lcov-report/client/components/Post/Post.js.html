<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Post/Post.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/Post</a> Post.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/7</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import styles from './post.css';
import {connect} from 'react-redux';
import { removePost } from "../../actions/actions";
import {withRouter} from 'react-router-dom';
import { Link } from 'react-router-dom'
&nbsp;
class Post extends React.Component{
    constructor(props){
        super(props);
&nbsp;
    }
&nbsp;
    render(){
        const info = this.props.info;
        return (
            &lt;article className="poster"&gt;
                &lt;h3 className="title"&gt;{info.title}&lt;/h3&gt;
                &lt;input type="button" value="Delete" className="delete-button-post" 
                onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; {<span class="cstat-no" title="statement not covered" >this.props.fetchRemovePosts(info.id)}</span>}/&gt;
                &lt;p className="description"&gt;{info.description}&lt;/p&gt;
                &lt;div className="post-inform"&gt;
                    &lt;span className="data"&gt;{info.releaseDate}&lt;/span&gt;
                    &lt;span className="author"&gt;{info.author}&lt;/span&gt;
                &lt;/div&gt;
            &lt;/article&gt;
        );
    }
}
&nbsp;
const mapDispatchToProps = (dispatch) =&gt; {
    return {
      fetchRemovePosts: <span class="fstat-no" title="function not covered" >(i</span>d) =&gt; {
<span class="cstat-no" title="statement not covered" >          dispatch(removePost(id))</span>
      }
  };
}
&nbsp;
export default withRouter(connect(null,mapDispatchToProps)(Post));;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 04 2018 09:13:44 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
