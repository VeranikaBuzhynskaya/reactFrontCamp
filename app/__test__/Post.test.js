import React from 'react';
import { mount, shallow } from 'enzyme';
import { Link, BrowserRouter} from 'react-router-dom';
import { Provider } from "react-redux";
import Post from '../client/components/Post/Post';
import renderer from 'react-test-renderer';

import data from './mockOnePostData';
import storeConstructor from '../client/store';

const store = storeConstructor({});

describe('Post', () => {
  let blogPost;

  beforeAll(() => {
    blogPost = mount(
        <BrowserRouter>
          <Provider store={store}>
                <Post info ={data} key={data.id}/>
          </Provider>
        </BrowserRouter>
    );
    debugger;
    console.log('blogPost', <Post info ={data}/> );
  })

  it('has correct title', () => {
    debugger;
    expect(blogPost.find('.poster h3').first().text())
        .toEqual("React Design Principles");
  });

  it('has correct description', () => {

    expect(blogPost.find('.description').text())
        .toEqual("Thriller Thriller Thriller Thriller Thriller");
  });

  it('has correct release date', () => {

    expect(blogPost.find('.post-inform span').first().text())
      .toEqual("2015");
  });

  it('has correct author name', () => {
    
        expect(blogPost.find('.post-inform span').last().text())
          .toEqual("Dan Abramov");
      });

  it('matches snapshot for components Post', () =>{
    const tree = renderer.create(
      <BrowserRouter>
      <Provider store={store}>
            <Post info ={data} key={data.id}/>
      </Provider>
    </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
})