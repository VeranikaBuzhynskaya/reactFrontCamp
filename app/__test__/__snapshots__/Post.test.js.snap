// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches snapshot for components Post 1`] = `
<article
  className="poster"
>
  <h3
    className="title"
  >
    React Design Principles
  </h3>
  <input
    className="delete-button-post"
    onClick={[Function]}
    type="button"
    value="Delete"
  />
  <p
    className="description"
  >
    Thriller Thriller Thriller Thriller Thriller
  </p>
  <div
    className="post-inform"
  >
    <span
      className="data"
    >
      2015
    </span>
    <span
      className="author"
    >
      Dan Abramov
    </span>
  </div>
</article>
`;
